File: Lab Week 1 Part 1/java/Inheritance.java
public class Inheritance {
    static class Animal {
        protected String name;
        public Animal(String name) { this.name = name; }
        public void speak() {
            System.out.println(name + " makes a sound");
        }
    }

    static class Dog extends Animal {
        public Dog(String name) { super(name); }
        @Override
        public void speak() {
            System.out.println(name + " barks");
        }
    }

    public static void run() {
        Animal a = new Animal("Generic");
        Dog d = new Dog("Buddy");
        a.speak();
        d.speak();
    }
}

File: Lab Week 1 Part 1/dart/bin/inheritance.dart
class Animal {
  final String name;
  Animal(this.name);

  void speak() {
    print('$name makes a sound');
  }
}

class Dog extends Animal {
  Dog(String name) : super(name);

  @override
  void speak() {
    print('$name barks');
  }
}

void run() {
  var a = Animal('Generic');
  var d = Dog('Buddy');
  a.speak(); // Generic makes a sound
  d.speak(); // Buddy barks
}